### PR TITLE
format: remove trailing whitespace from bound comments

### PR DESCRIPTION
**Pull request by V-FEXrt**
Fixes trailing  white space that is  emitted after a  nested floating comment. Also adds more tests around floating  comments to catch other potential breakages

### CODE REVIEW

**Review by ag-eitilt — COMMENTED** [review]
Good catch!  Do we have anything removing end-of-line spacing?  (I'm pretty sure that's taken care of by default for the actual code, but we might want to strip within comments as well.)

File: tools/wake-format/actions.h [hunk 1]
```diff
@@ -121,7 +143,29 @@ struct TokenReplaceAction {
     auto it = traits.find(node);
     if (it != traits.end()) {
       for (auto n : it->second.before_bound) {
-        builder.append(n.fragment().segment().str());
+        if (n.id() == TOKEN_COMMENT) {
```
Comment (JakeSiFive) [inline tools/wake-format/actions.h:146]: I'm going to pull an Ashley and ask if we can factor out the repeated code between the repeated code here. I'm happy to not do it though as well.

**Review by JakeSiFive — APPROVED** [review]
C++ code looks good to me

File: tools/wake-format/actions.h [hunk 2]
```diff
@@ -121,12 +91,38 @@ struct TokenReplaceAction {
...
+      }
+
+      // Realign indent in case we lost alighnment.
+      if (it->second.before_bound.size() > 0) {
         freshline(builder, ctx);
       }
     }
 
-    builder.append(str);
+    if (str == nullptr) {
```
Comment (JakeSiFive) [inline tools/wake-format/actions.h:121]: sick, like this solution!

**Review by JakeSiFive — APPROVED** [review]

**Pull request merged**